rust/xaynet-core/src/mask/masking.rs

```diff
@@ -190,18 +188,26 @@ impl Aggregation {
     ///
     /// [`validate_unmasking()`]: #method.validate_unmasking
     /// [`mask()`]: struct.Masker.html#method.mask
-    pub fn unmask(mut self, mask_obj: MaskObject) -> Model {
-        // unmask (over-scaled) global model
-        let scaled_add_shift_n =
-            self.object.vector.config.add_shift() * BigInt::from(self.nb_models);
-        let exp_shift_n = self.object.vector.config.exp_shift();
-        let order_n = self.object.vector.config.order();
-        let model: Model = self
-            .object
-            .vector
+    pub fn unmask(self, mask_obj: MaskObject) -> Model {
```

Comment:
we might destructure `self` and `mask_obj` before using it below, but that's mostly a matter of taste and up to your decision (i also removed the notion of `weight` and updated the `panic_safe` comment):
```rust
pub fn unmask(self, mask_obj: MaskObject) -> Model {
    let Aggregation {
        nb_models,
        object:
            MaskObject {
                vect:
                    MaskVect {
                        data: masked_n,
                        config: config_n,
                    },
                unit:
                    MaskUnit {
                        data: masked_1,
                        config: config_1,
                    },
            },
        ..
    } = self;
    let MaskObject {
        vect: MaskVect { data: mask_n, .. },
        unit: MaskUnit { data: mask_1, .. },
    } = mask_obj;

    // unmask scalar sum
    let scaled_add_shift_1 = config_1.add_shift() * BigInt::from(nb_models);
    let exp_shift_1 = config_1.exp_shift();
    let order_1 = config_1.order();
    let n = (masked_1 + &order_1 - mask_1) % &order_1;
    let ratio = Ratio::<BigInt>::from(n.to_bigint().unwrap());
    let scalar_sum = ratio / &exp_shift_1 - &scaled_add_shift_1;

    // unmask global model
    let scaled_add_shift_n = config_n.add_shift() * BigInt::from(nb_models);
    let exp_shift_n = config_n.exp_shift();
    let order_n = config_n.order();
    masked_n
        .into_iter()
        .zip(mask_n)
        .map(|(masked, mask)| {
            // PANIC_SAFE: The substraction panics if it
            // underflows, which can only happen if:
            //
            //     mask > order_n
            //
            // If the mask is valid, we are guaranteed that this
            // cannot happen. Thus this method may panic only if
            // given an invalid mask.
            let n = (masked + &order_n - mask) % &order_n;

            // UNWRAP_SAFE: to_bigint never fails for BigUint
            let ratio = Ratio::<BigInt>::from(n.to_bigint().unwrap());
            let unmasked = ratio / &exp_shift_n - &scaled_add_shift_n;

            // scaling correction
            unmasked / &scalar_sum
        })
        .collect()
}
```

Reply:
nice! I've destructured like so
```rust
let (masked_n, config_n) = (self.object.vect.data, self.object.vect.config);
let (masked_1, config_1) = (self.object.unit.data, self.object.unit.config);
let mask_n = mask_obj.vect.data;
let mask_1 = mask_obj.unit.data;
```
to save a few lines, otherwise taken your suggestion.